src/openlcb/NonAuthoritativeEventProducer.hxx

```diff
@@ -0,0 +1,207 @@
+ *    this list of conditions and the following disclaimer.
+ *
+ *  - Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @file NonAthoritativeEventProducer.hxx
+ *
+ * Defines implementations of Event Producers that are uathoratative, meaning
+ * that they do not represent the actual state of an event, but can still
+ * query that state (presumably from a consumer) and produce a new state.
+ *
+ * @author Stuart Baker
+ * @date 17 June 2017
+ */
+
+#ifndef _OPENLCB_NONAUTHORITATIVEEVENTPRODUCER_HXX_
+#define _OPENLCB_NONAUTHORITATIVEEVENTPRODUCER_HXX_
+
+#include "openlcb/EventHandlerTemplates.hxx"
+
+namespace openlcb
+{
+#if 0
+    DEFPROXYFN(handle_event_report);
+    DEFPROXYFN(handle_consumer_identified);
+    DEFPROXYFN(handle_consumer_range_identified);
+    DEFPROXYFN(handle_producer_identified);
+    DEFPROXYFN(handle_producer_range_identified);
+    DEFPROXYFN(handle_identify_global);
+    DEFPROXYFN(handle_identify_consumer);
+    DEFPROXYFN(handle_identify_producer);
+
+    IGNOREFN(handle_event_report);
+    IGNOREFN(handle_consumer_identified);
+    IGNOREFN(handle_consumer_range_identified);
+    IGNOREFN(handle_producer_identified);
+    IGNOREFN(handle_producer_range_identified);
+    IGNOREFN(handle_identify_consumer);
+    IGNOREFN(handle_identify_producer);
+#endif
+
+/// Event producer for range of bits (event pairs) that is non-autoritative.
+class BitRangeNonAuthoritativeEventP : public SimpleEventHandler
+{
+public:
+    /// Constructor.  Creates a new bit range producer.
+    ///
+    /// @param node the node that the producer will be bound to
+    /// @param event_base the starting event ID for the event range
+    /// @param size The total event range size in "pairs" of events.  Each
+    ///             pair of sequential events represents a single "bit" with
+    ///             a binary state.
+    /// @param state_callback Callback method for delivering the results of a
+    ///                       consumer identified.  The first unsigned parameter
```

Comment:
The code already calls the same callback for both "event report" and "consumer identified".  The comment should be updated accordingly.